packages/babel-plugin-transform-es2015-classes/src/vanilla.js

```diff
@@ -25,15 +25,20 @@ const noMethodVisitor = {
 };
 
 const verifyConstructorVisitor = visitors.merge([noMethodVisitor, {
-  Super(path) {
+  Super(path, state) {
     if (
       this.isDerived && !this.hasBareSuper &&
-      !path.parentPath.isCallExpression({ callee: path.node })
+      !path.parentPath.isCallExpression({ callee: path.node }) &&
+      !state.inArrowFunctionExpression
     ) {
       throw path.buildCodeFrameError("'super.*' is not allowed before super()");
     }
   },
 
+  ArrowFunctionExpression(path, state) {
+    state.inArrowFunctionExpression = true;
```

Comment:
Since this is never cleared, I think this will not error for a case like

    var t = () => super.test();
    super.foo();
    super();

The flag will be set, and then once we traverse out of the arrow, it will still be set.

Potentially we can check

     path.findParent((p) => p.isFunction()).isArrowFunctionExpression()

?